olive/evaluator/olive_evaluator.py

```diff
@@ -823,8 +822,7 @@ def _inference(
         device: Device = Device.CPU,
         execution_providers: Union[str, List[str]] = None,
     ) -> Tuple[OliveModelOutput, Any]:
-        session = model.prepare_session(inference_settings=self.get_inference_settings(metric), device=device)
-
+        session = model.prepare_session()
```

Comment:
>prepare_session [](http://example.com/codeflow?start=24&length=15)

why do we remove the device from the prepare_session?

Reply:
We enable session device here: https://github.com/microsoft/Olive/blob/main/olive/evaluator/olive_evaluator.py#L833. We could refactor this to move this to `prepare_session`